src/file_operations_test.js

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env node
+
+const path = require("path")
+const { assert } = require("chai")
+const fileOperations = require("./file_operations")
+const markdownGenerators = require("./markdown_generators")
+const fs = require("fs")
+const sinon = require("sinon")
+const tmp = require("tmp")
+tmp.setGracefulCleanup()
+
+const singleCourseId =
+  "1-00-introduction-to-computers-and-engineering-problem-solving-spring-2012"
+const singleCourseSourcePath = `test_data/${singleCourseId}`
+const singleCourseMasterJsonPath = path.join(
+  singleCourseSourcePath,
+  "bb55dad7f4888f0a1ad004600c5fb1f1_master.json"
+)
+const singleCourseRawData = fs.readFileSync(singleCourseMasterJsonPath)
+const singleCourseJsonData = JSON.parse(singleCourseRawData)
+const singleCourseMarkdownData = markdownGenerators.generateMarkdownFromJson(
+  singleCourseJsonData
+)
+
+describe("scanCourses", () => {
+  let readdirSync, lstatSync, consoleLog
+  const sandbox = sinon.createSandbox()
+  const sourcePath = "test_data"
+  const destinationPath = tmp.dirSync({ prefix: "destination" }).name
+  const logMessage = "Scanning 3 subdirectories under test_data"
+  const course1Path = path.join(
+    sourcePath,
+    "1-00-introduction-to-computers-and-engineering-problem-solving-spring-2012"
+  )
+  const course2Path = path.join(
+    sourcePath,
+    "2-00aj-exploring-sea-space-earth-fundamentals-of-engineering-design-spring-2009"
+  )
+  const course3Path = path.join(
+    sourcePath,
+    "3-00-thermodynamics-of-materials-fall-2002"
+  )
+
+  beforeEach(() => {
+    readdirSync = sandbox.spy(fs, "readdirSync")
+    lstatSync = sandbox.spy(fs, "lstatSync")
+    consoleLog = sandbox.stub(console, "log")
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it("throws an error when you call it with no source directory", () => {
+    try {
+      fileOperations.scanCourses(null, destinationPath)
+    } catch (err) {
+      assert.equal(err.message, "Invalid source directory")
+    }
```

Comment:
you might be able to use the chai `assert.throws` method for these, although what you have here works too